databroker/pims_readers.py

```diff
@@ -44,7 +44,8 @@ def __init__(self, headers, name):
                 # do something
         """
         events = get_events(headers, [name], fill=False)
-        self._datum_uids = [event.data[name] for event in events]
+        self._datum_uids = [event.data[name] for event in events
+                            if name in event.data]
         self._len = len(self._datum_uids)
```

Comment:
if `_datum_uids` is an empty list here would it make sense to raise and let the user know that they gave a bad key?